lib/PuppeteerSharp.Tests/FrameTests/WaitForFunctionTests.cs

```diff
@@ -128,5 +128,22 @@ public async Task ShouldDisableTimeoutWhenItsSetTo0()
             await Page.EvaluateExpressionAsync("window.__injected = true");
             await watchdog;
         }
+
+        [Fact]
+        public async Task ShouldSurviveCrossProcessNavigation()
+        {
+            var fooFound = false;
+            var waitForFunction = Page.WaitForFunctionAsync("window.__FOO === 1")
```

Comment:
Puppeteer has WaitForFunction and WaitForExpression in only one function `waitForFunction`.
You should use `WaitForExpression` here.